nilearn/_utils/niimg_conversions.py

```diff
@@ -44,6 +91,10 @@ def check_niimg(niimg, atleast_4d=False):
         call nibabel.load on it. If it is an object, check if get_data()
         and get_affine() methods are present, raise TypeError otherwise.
 
+    ndim: integer {3, 4}, optional
+        Indicate the dimensionality of the expected niimg. An
+        error is raised if the niimg is of another dimensionality.
+
     atleast_4d: boolean, optional
         Indicates if a 3d image should be turned into a single-scan 4d niimg.
```

Comment:
I would be more in favor of "ensure_ndim". And no specific objection for "coerce_4d".